### PR TITLE
Fix a potential crash in allies05a

### PR DESCRIPTION
**Pull request by abcdefg30**
Pretty much the same as #7375.

### CODE REVIEW

File: mods/ra/maps/allies-05a/AI.lua [hunk 1]
```diff
@@ -61,8 +61,8 @@ ProtectHarvester = function(unit)
 
 		local Guards = SetupAttackGroup()
 		Utils.Do(Guards, function(unit)
-			if attacker.Location then
-				unit.AttackMove(attacker.Location)
+			if not self.IsDead then
```
Comment (obrakmann) [inline mods/ra/maps/allies-05a/AI.lua]: Shouldn't you check if `unit` is dead, too?

Comment (abcdefg30) [inline mods/ra/maps/allies-05a/AI.lua]: `SetupAttackGroup` should already do that: https://github.com/abcdefg30/OpenRA/blob/protectharvester/mods/ra/maps/allies-05a/AI.lua#L28

Comment (obrakmann) [inline mods/ra/maps/allies-05a/AI.lua]: Oh, alright then.

**Comment by obrakmann** [comment]
:+1:

**Comment by Mailaender** [comment]
It also seems to make more sense to rush to the harvester. :+1:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/a6a28470f394e6f367ac3c9c155c9e9091a9c2d9...544aac3e0dddd730e499a6ac130efb86518b18f8)

**Pull request merged**